src/Engine/EmitMSIL/Replication.cs

```diff
@@ -2,17 +2,107 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Reflection;
 using ProtoCore;
-using ProtoCore.Utils;
 using ProtoCore.Lang.Replication;
+using ProtoCore.Exceptions;
+using ProtoCore.Properties;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
 
 namespace EmitMSIL
 {
     public class Replication
     {
+        internal static List<CLRStackValue> MarshalFunctionArguments(IList args, MSILRuntimeCore runtimeCore)
```

Comment:
how would you describe what these marshal methods are doing? What are they marshaling between, since we now have a compiler instead of a VM? Or does this PR bring back a lot of a the machinery of the VM?

Asked another way, why do we need the runtime core, class table etc?

Reply:
We are marshaling between the pure CLR objects and the CLRStackValue structure.
We need the CLRStackValue, runtime core and class table only for performance reasons (and maybe to simplify the copy/paste process).

We need type information (that can represent CLR and DesingScript types - ex `var`), rank, coercions scores etc at runtime. Having this information pre-processed and stored at compile time ...will improve performance at runtime.

Reply:
At the moment I am doing most of this work at runtime...just to avoid the extra work of doing it at compile time.
But a lot of the performance gain comes when we pre-process this info at compile time